.github/workflows/pre-commit.yml

```diff
@@ -8,8 +8,38 @@ jobs:
   pre-commit:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-python@v4
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
todo replace with https://github.com/getsentry/action-pre-commit, i'll do this on monday